src/image_processing/processing.py

```diff
@@ -0,0 +1,131 @@
+            self.last_class_predictions.append(class_prediction)
+            self.last_box_predictions.append(box_prediction)
+        else:
+            # Delete first element from queue
+            self.last_class_predictions.pop(0)
+            self.last_box_predictions.pop(0)
+            # Add new element to queue
+            self.last_class_predictions.append(class_prediction)
+            self.last_box_predictions.append(box_prediction)
+
+    def is_queue_full(self):
+        if len(self.last_class_predictions) == self.queue_size and len(self.last_box_predictions) == self.queue_size:
+            return True
+        else:
+            return False
+
+    def calculate_drawing_state(self):
+        """
+        Calculates drawing state based on last predictions.
+
+        self.last_class_predictions is a queue which stores last class predictions (0 for fist <stop signal> and 1
+        for palm <start signal>). If the mean from queue is less than drawing_state_threshold and the hand didn't move
+         it disables the drawing.
+
+        :return:
+        """
+        if self.is_queue_full():
+            # Compute mean of last class predictions. If closer to 0, then it's bigger chance that the Fist occured. If
+            # closer to 1, then it's bigger chance that the Palm occured.
+            class_predictions_mean = np.mean(self.last_class_predictions)
+
+            # Compute standard deviation in last box positions
+            last_box_predictions_array = np.array(self.last_box_predictions)
+            box_std_dev = np.mean(np.std(last_box_predictions_array, axis=0))
+
+            if (class_predictions_mean < self.drawing_state_threshold) \
+                    and (box_std_dev < self.inactivity_std_dev_threshold):
+                if self.drawing_state and not self.is_outlier:
+                    # This is the case when fist is not moving, and previously state was drawing state
+                    self.finish_drawing = True
+                # This is the case when Fist is not moving
+                self.drawing_state = False
+
+            elif class_predictions_mean > self.drawing_state_threshold:
+                if self.activity_std_dev_lower_threshold < box_std_dev < self.activity_std_dev_upper_threshold:
+                    # This is the case when Palm is moving normally
+                    self.drawing_state = True
+                elif box_std_dev > self.activity_std_dev_upper_threshold:
+                    # This is the case when Palm had a huge move. We classify this as a outlier
+                    self.is_outlier = True
+
+    def normalize_drawing_points(self):
+        self.drawing_points = np.array(self.drawing_points, dtype=np.float)
+        x_max = np.max(self.drawing_points[:, 0])
+        self.drawing_points[:, 0] = self.drawing_points[:, 0] / x_max
+
+        y_max = np.max(self.drawing_points[:, 1])
+        self.drawing_points[:, 1] = self.drawing_points[:, 1] / y_max
```

Comment:
Why do you normalize the points to the maximum values ​​that occurred and not to the image width/height ?

Reply:
Because when I normalize this by using maximum values that occurred I make it easier to control later on ( For example if a person draws a very little drawing, then the values will have a very small spread, which will be almost impossible to repeat by drone), and I get rid of useless information where the drawing occurred on the image (for example if someone drew a  pattern in a bottom right corner then it's projected to the whole image, rather than bottom right corner.).

Reply:
All right